src/cli/common.rs

```diff
@@ -147,7 +148,10 @@ impl NotifyOnConsole {
     }
 }
 
-pub fn set_globals(verbose: bool, quiet: bool) -> Result<Cfg> {
+pub fn set_globals<V>(vars: &V, verbose: bool, quiet: bool) -> Result<Cfg<V>>
+where
+    V: VarSource + Clone,
```

Comment:
Given how often you say `VarSource + Clone` would it not be easier to just supertrait it?  If we ever need to make a non-Clone VarSource we could revisit it.

Reply:
By this point I was doing large scale grep-replace and not thinking :) - I only found out about the Clone need when I had the entire patch in place and was substituting in the HashMap layer in the WIP branch and found it was missing. Sure. I think perhaps the refactoring I refer to in the conversation is perhaps a bigger better rearrangement.